modules/apps/29-fee/ibc_module.go

```diff
@@ -130,8 +130,20 @@ func (im IBCModule) OnChanCloseInit(
 	portID,
 	channelID string,
 ) error {
-	// TODO: Unescrow all remaining funds for unprocessed packets
+	// delete fee enabled on channel
+	// and refund any remaining fees escrowed on channel
 	im.keeper.DeleteFeeEnabled(ctx, portID, channelID)
+	err := im.keeper.RefundFeesOnChannel(ctx, portID, channelID)
+	// error should only be non-nil if there is a bug in the code
+	// that causes module account to have insufficient funds to refund
+	// all escrowed fees on the channel.
+	// Disable all channels to allow for coordinated fix to the issue
+	// and mitigate/reverse damage.
+	// NOTE: Underlying application's packets will still go through, but
+	// fee module will be disabled for all channels
+	if err != nil {
+		im.keeper.DisableAllChannels(ctx)
```

Comment:
Disabling all channels will effectively "turn off" the fee module. I added a check for `IsFeeEnabled` on `EscrowFee`.

Also, `OnAckPacket` and `OnTimeoutPacket` will check if fee is enabled on the channel before running middleware logic. So if we disable all channels, any further escrows will get rejected and all distributions will be bypassed. Thus, we can turn off the fee functionality without affecting underlying applications. IBC can still run even if fee module gets disabled due to invalid state

Reply:
It would be nice to have `IsFeeEnabled` as a param we can change via governance. That way if a bug was discovered a chain could potentially shut down a channel. Although, maybe the proposal would take too long. 

Either way, I think this approach works better. Nice :+1:

Reply:
Potentially we could have a global enabled param, rather than a per-channel one. But that should be in a separate PR

Reply:
Agreed :+1: